app/controllers/middleware_server_controller.rb

```diff
@@ -273,22 +292,26 @@ def run_server_operation(operation_info, items)
   end
 
   def trigger_mw_operation(operation, mw_server, params = nil)
-    mw_manager = mw_server.ext_management_system
-    path = mw_server.ems_ref
+    if operation == :generate_jdr
```

Comment:
I don't like this, it's adding a conditional on a method supposed to add genericity to the operation. Either bypass it on the caller or extract this into something else.